packages/strapi-plugin-graphql/services/build-aggregation.js

```diff
@@ -252,19 +270,52 @@ const createGroupByFieldsResolver = function(model, fields) {
       ...convertToQuery(filters.where),
     };
 
-    const result = await buildQuery({
-      model,
-      filters: convertRestQueryParams(params),
-      aggregate: true,
-    }).group({
-      _id: `$${fieldKey === 'id' ? model.primaryKey : fieldKey}`,
-    });
+    if (model.orm === 'mongoose') {
+      const result = await buildQuery({
+        model,
+        filters: convertRestQueryParams(params),
+        aggregate: true,
+      }).group({
+        _id: `$${fieldKey === 'id' ? model.primaryKey : fieldKey}`,
+      });
+
+      return preProcessGroupByData({
+        result,
+        fieldKey,
+        filters,
+      });
+    }
+
+    if (model.orm === 'bookshelf') {
+      let { limit, ...aggFilters } = filters;
+      return model
+        .query(qb => {
+          // apply filters without pagination limit
+          buildQuery({ model, filters: aggFilters })(qb);
+          qb.groupBy(fieldKey);
+          qb.select(fieldKey);
+        })
+        .fetchAll()
+        .then(result => {
```

Comment:
you can do `m.get(fieldKey)` instead of getting it throught the internal model structure

Reply:
did you mean model for the `m` in `m.get(fieldKey)`?